packages/bundler-plugin-core/src/index.ts

```diff
@@ -455,7 +485,23 @@ const unplugin = createUnplugin<Options>((options, unpluginMetaContext) => {
         });
       }
     },
-  };
+  });
+
+  if (unpluginMetaContext.framework === "esbuild") {
+    if (internalOptions._experiments.debugIdUpload) {
+      plugins.push({
+        name: "sentry-debug-id-plugin",
```

Comment:
> shouldn't we specifically highlight this is esbuild only?

I don't think it is necessary to say it is explicitly for esbuild, since it will only be used in esbuild anyways, but this is such a little change I will just do it :p

> Or we can also move the rest of the debug id logic into this plugin.

Unfortunately, this is a bit of a limitation/bug with unplugin itself. You cannot define hooks + a `config` field in the same plugin, so we need to emit two. If you define hooks + `config`, the hooks will not run.